### DISCUSSION TITLE
New translation in Pidgin (West Africa)

### DISCUSSION THREAD

**Discussion by mynetx** [sircharlo/meeting-media-manager#720] [Translations]
I would like to help to translate M³ into a language I speak,Pidgin (West Africa) (PGW/wes-x-pgw).

**Comment by mtdvlpr** [comment]
Thank you so much!

To get started, please [sign into GitLocalize](https://gitlocalize.com/) using your GitHub account, and then let me know when that's done by adding a comment in this chat. 

As language moderator, you will get notified of changes to the English original strings, have a nice and easy to use interface to review existing as well as untranslated strings in your language, and submit new translations to the repo easily.

**Comment by mynetx** [comment]
I've registered on GitLocalize. :)

**Comment by mtdvlpr** [comment]
Gitlocalize doesn't have your specific language, so I put you down under Afrikaans / African :)

**Reply by mtdvlpr** [reply]
Is this accurate or is there another language that it is most similar to?

**Reply by mynetx** [reply]
The language is not similar to Afrikaans (which is spoken in South Africa). Pidgin English is spoken in Nigeria and Cameroon.

https://en.wikipedia.org/wiki/West_African_Pidgin_English

**Reply by mtdvlpr** [reply]
Sorry about that! I have made [an issue at Gitlocalize](https://github.com/gitlocalize/feedback/issues/121) to add your language. In the meantime Gitlocalize has the following options for me to select from. Which of these would you say is most fitting?
<details><summary> Open list</summary>

Spanish [es]

German [de]

French [fr]

Portuguese [pt]

Russian [ru]

Chinese [zh]

Acholi [ach]

Afrikaans [af]

Albanian [sq]

Amharic [am]

Arabic [ar]

Armenian [hy]

Azeri [az]

Basque [eu]

Belarusian [be]

bengali [bn]

Bosnian [bs]

Bulgarian [bg]

Burmese [my]

Catalan [ca]

Cebuano [ceb]

Khmer [km]

Nyanja [ny]

Chinese, China [zh-CN]

Chinese, China [zh-cn] [zh-cn]

Chinese, China [zh_CN] [zh_CN]

Chinese, Taiwan [zh-TW]

Corsican [co]

Croatian [hr]

Czech [cs]

Czech, Czech Republic [cs-CZ]

Danish [da]

Dutch [nl]

Dutch, Netherlands [nl-NL]

English, United Kingdom [en-GB]

English, United States [en-US]

Esperanto [eo]

Estonian [et]

Ewe [ee]

Finnish [fi]

French, France [fr-FR]

Scottish Gaelic [gd]

Galician [gl]

Georgian [ka]

German, Germany [de-DE]

Greek [el]

Gujarati [gu]

Haitian [ht]

Hausa [ha]

Hawaiian [haw]

Hebrew [he]

Hindi [hi]

Hmong [hmn]

Hungarian [hu]

Icelandic [is]

Igbo [ig]

Indonesian [id]

Irish [ga]

Italian [it]

Italian, Italy [it-IT]

Japanese [ja]

Japanese, Japan [ja-JP]

Kannada [kn]

Kazakh [kk]

Kirghiz [ky]

Korean [ko]

Korean, South Korea [ko-KR]

Kurdish [ku]

Lango [laj]

Lao [lo]

Latin [la]

Latvian [lv]

Lithuanian [lt]

Luganda [lug]

Luxembourgish [lb]

Macedonian [mk]

Malagasy [mg]

Malay [ms]

Malayalam [ml]

Maltese [mt]

Manipuri [mni]

Maori [mi]

Marathi [mr]

Mongolian [mn]

Nepali [ne]

Nkore [nun]

Nagamese [nag]

Norwegian [no]

Norwegian Bokmål [nb]

Norwegian Nynorsk [nn]

Oriya [or]

Persian [fa]

Persian, Iran [fa-IR]

Polish [pl]

Portuguese, Brazil [pt-br]

Portuguese, Brazil [pt-BR]

Portuguese, Portugal [pt-PT]

Portuguese, Brazil [pt_BR] [pt_BR]

Punjabi [pa]

Pushto [ps]

Romanian [ro]

Romanian, Romania [ro-RO]

Rundi [rn]

Samoan [sm]

Serbian [sr]

Serbian, Serbia [sr-RS]

Shona [sn]

Sindhi [sd]

Sinhala [si]

Slovak [sk]

Slovak, Slovakia [sk-SK]

Slovenian [sl]

Somali [so]

Southern Sotho [st]

Spanish, Spain [es-ES]

Sundanese [su]

Swahili [sw]

Swedish [sv]

Swedish, Sweden [sv-SE]

Tagalog [tl]

Tajik [tg]

Tamil [ta]

Telugu [te]

Thai [th]

Turkish [tr]

Ukrainian [uk]

Urdu [ur]

Uzbek [uz]

Vietnamese [vi]

Welsh [cy]

Western Frisian [fy]

Xhosa [xh]

Soga [xog]

Yiddish [yi]

Yoruba [yo]

Zulu [zu]



</details>

**Reply by mynetx** [reply]
Thank you for that list, my brother! But the thing is, West African Pidgin English (JW combined Nigerian and Cameroonian Pidgin for easier translation, because they only differ minimally) doesn't resemble any of these languages. The closest language would be "English" :D

**Reply by mtdvlpr** [reply]
Alright, would you prefer I set your language to English, United Kingdom [en-GB] then? It's only of consequence inside Gitlocalize. In the app I will set the correct language name for your translations.

**Reply by mynetx** [reply]
Okay, good. You'll see the ISO code on that website ⬆️  and the organisation code is PGW. Can we move those strings to the correct locale once that issue is resolved? I'm not sure how fast they'll be.

**Reply by mtdvlpr** [reply]
Yes, of course! The language code is customizable, so that's now wes-x-pgw (in accordance with jw.org).

**Reply by mtdvlpr** [reply]
Is the ISO code `cpe-011` then?

**Reply by mynetx** [reply]
The language doesn't have an ISO code, because it's a JW mix of Nigerian and Cameroonian. It only has an IETF code:

<img width="232" alt="image" src="https://user-images.githubusercontent.com/6348321/202686386-4037d48b-e92a-454c-809f-7c1f391f8267.png">

**Comment by mtdvlpr** [comment]
I have added you to both gitLocalize repos. The sircharlo one is for the actual application (most important). The mtdvlpr one is for the documentation website (nice to have). Thank you so much for your willing spirit!

**Reply by mynetx** [reply]
@mtdvlpr So there are two identically named repos but in different GitHub users and with different purposes? I also saw there's a revamp in the works?

**Reply by mtdvlpr** [reply]
Yes, the revamp has been finished and is version 2.9.0. That has been made in my forked repo (mtdvlpr/meeting-media-manager). The documentation site is hosted through the orphaned docs branch. The problem is that Gitlocalize doesn't support translating multiple branches of the same repo, so the fix is to use my forked repo as a translation repo for the documentation site.

**Reply by mtdvlpr** [reply]
I see you're a developer yourself, so if you have any suggested improvements on this or other technologies we use, feel free to mention them!

**Comment by mynetx** [comment]
#725

**Comment by mtdvlpr** [comment]
@mynetx, could you answer this for me? 

https://github.com/gitlocalize/feedback/issues/121#issuecomment-1320441332

**Comment by mtdvlpr** [comment]
@mynetx, you are now listed as the moderator of Nigerian Pidgin in Gitlocalize :)

**Comment by mtdvlpr** [comment]
Hi @mynetx, for local dates we use dayjs which does not support Nigerian Pidgin. Would you be willing to contribute to their package and add Nigerian Pidgin to their locale? It's not a lot, just some date/time translations, like the days of the week. You can add an issue at their repo [here](https://github.com/iamkun/dayjs/issues). The fallback is English, so you only have to do this if you want the months on the home screen and the day in the localeDateFormat to be in Nigerian Pidgin as well. Feel free to leave it as is. Just wanted to let you know how you could translate those strings :)

**Comment by mynetx** [comment]
Well, they use the same days and months as in English. Would you still suggest to add it?

**Reply by mtdvlpr** [reply]
No, just leave it then.
